src/circuit.rs

```diff
@@ -305,18 +321,32 @@ where
     /// instances.
     fn gen_proof(
         &mut self,
-        pub_params: &PublicParameters<E>,
+        u_params: &UniversalParams<E>,
         prover_key: ProverKey<E::Fr, P>,
         transcript_init: &'static [u8],
     ) -> Result<Proof<E, P>, Error> {
-        let (ck, _) = pub_params.trim(self.padded_circuit_size())?;
+        // XXX: KZG10 does not have a trim function so we use sonics and
+        // then do a transformation between sonic CommiterKey to KZG10
```

Comment:
Small nit `from` sonic may be better

Reply:
I think it is not possible to implement a foreign trait between foreign structs. However it may be a good idea to create a wrapper for the commiter keys and implement the transformations there